mapper/orm/src/main/java/org/hibernate/search/mapper/orm/search/loading/impl/HibernateOrmByTypeEntityLoader.java

```diff
@@ -47,7 +48,7 @@
 				referencesByDelegate.entrySet() ) {
 			HibernateOrmComposableEntityLoader<? extends T> delegate = entry.getKey();
 			List<EntityReference> referencesForDelegate = entry.getValue();
-			delegate.loadBlocking( referencesForDelegate, objectsByReference );
+			delegate.loadBlocking( referencesForDelegate, objectsByReference, timeout );
```

Comment:
That's wrong, though. If there are 5 entity types and a timeout of 1ms, you're allowing each entity type to take 1ms, so in the end you allow 5ms.

To address this, you shouldn't pass a timeout, you should pass a deadline (Instant or long representing the nanos, whatever you prefer).

Reply:
I had thought of passing the new duration net of the time elapsed, but it is not yet implemented in this commit. I would use the duration to be consistent with the main API, but if you prefer a long with the remaining nanoseconds or an instant for the deadline, it's the same.

Reply:
Converting to an `Instant` just after the API and then using always the same util to get the remaining duration, everywhere, would seem more fool-proof to me. But as long as it works...